matchms/similarity/BaseSimilarity.py

```diff
@@ -28,87 +30,126 @@ class BaseSimilarity:
 
     def __init__(self, score_filters: Tuple[FilterScoreByValue, ...]):
         """
-
-        Attributes
+        Parameters
         ----------
         score_filters:
-            Tuple of filters that should be applied to each score before scoring. If you want to run without filtering,
-            it is best to run matrix(). Filters can also be run after computing a matrix first, however, for strict
-            filtering this implementation can be more memory efficient. Only use matrix_with_filter if you expect to
-            filter out more than 90% of your data. Otherwise, it is more memory efficient to first run matrix, followed by
-            filtering the matrix.
+            A tuple of filter objects to apply to each similarity score.
+            - If you do not wish to apply any filtering, pass an empty tuple and use the dense matrix method (`matrix()`).
+            - For cases where you expect to filter out more than 90% of the computed scores, consider using
+              the sparse array methods (see `sparse_array()`).
         """
         self.score_filters = score_filters
 
     @abstractmethod
     def pair(self, reference: SpectrumType, query: SpectrumType) -> np.ndarray:
-        """Method to calculate the similarity for one input pair.
+        """
+        Compute the similarity score for a single pair of spectra.
 
         Parameters
         ----------
         reference
-            Single reference spectrum.
+            A single reference spectrum.
         query
-            Single query spectrum.
+            A single query spectrum.
 
         Returns
-            The similarity score as numpy array (using self.score_datatype). For instance returning
-            np.asarray(score, dtype=self.score_datatype)
+            The similarity score as numpy array (with dtype given by self.score_datatype).
+            For example: return np.asarray(score, dtype=self.score_datatype)
         """
         raise NotImplementedError
 
-    def matrix(self, references: Sequence[SpectrumType],
-               queries: Sequence[SpectrumType],
-               is_symmetric: bool = False,
-               mask_indices: COOIndex = None,
-               ) -> np.ndarray:
+    def matrix(
+            self,
+            references: Sequence[SpectrumType],
+            queries: Sequence[SpectrumType],
+            is_symmetric: bool = False,
+            mask_indices: COOIndex = None) -> np.ndarray:
         """
         Compute a dense similarity matrix for all pairs of reference and query spectra.
 
+        Use this method when filtering is either required (with a provided mask) or when working
+        with moderately sized datasets. If no filtering is needed and you don't have a mask, simply
+        pass an empty tuple for `score_filters` and use this method.
+
```

Comment:
```suggestion
        Use this method when you expect most scores to pass the filtering. For instance if you don't apply any filtering. But also when you expect filtering to filter out less than 90% of all calculated scores. Filters can be added when creating the filter class object.
```

Reply:
This was a bit confusing to me, would this capture the same?

Reply:
Yes, should work, too. Only it is not clear what "filter class object" is I think.

Reply:
We should be very careful here because the word "filter" is also use for spectrum processing in matchms, so better make this as clear as possible.